client/login/controller.js

```diff
@@ -224,221 +212,201 @@ export function qrCodeLogin( context, next ) {
 	next();
 }
 
-export function googleAuth( context, next ) {
-	const { query, isServerSide } = context;
+export async function jetpackGoogleAuth( context, next ) {
+	const { query } = context;
+	const redirectUri = `https://${
+		config.isEnabled( 'oauth' ) ? window.location.host : 'wordpress.com'
+	}${ loginPath( { socialService: 'google' } ) }`;
 
-	// Skip processing on server-side render
-	if ( isServerSide ) {
+	try {
+		// Get authorization nonce for security
+		let nonce;
+		if ( config.isEnabled( 'oauth' ) ) {
+			// Use the API to get a real nonce in production
+			const response = await wpcomRequest( {
+				path: '/generate-authorization-nonce',
+				apiNamespace: 'wpcom/v2',
+				method: 'GET',
+			} );
+			nonce = response.nonce;
+		} else {
+			// In dev environment, use a random string as nonce
+			nonce = Math.random().toString( 36 ).substring( 2, 15 );
+		}
+
+		// Create state object with relevant data
+		const stateObject = {
+			redirect_to: query?.redirect_to || '/',
+			is_jetpack: true,
+			locale: context.params.lang,
+			wpcomNonce: nonce,
+			queryParams: { ...query },
+		};
+
+		// Store nonce in sessionStorage for validation on callback
+		window.sessionStorage.setItem( 'google_oauth_nonce', nonce );
+
+		// Load Google Identity Services API if not already loaded
+		if ( ! window?.google?.accounts?.oauth2 ) {
+			await loadScript( 'https://accounts.google.com/gsi/client' );
+			if ( ! window?.google?.accounts?.oauth2 ) {
+				throw new Error( 'Failed to load Google Identity Services API' );
+			}
+		}
+
+		// Initialize and request authorization code
+		window.google.accounts.oauth2
+			.initCodeClient( {
+				client_id: config( 'google_oauth_client_id' ),
+				scope: 'openid profile email',
+				ux_mode: 'redirect',
+				redirect_uri: redirectUri,
+				state: JSON.stringify( stateObject ),
+				callback: () => {},
+			} )
+			.requestCode();
+	} catch ( error ) {
+		/* eslint-disable-next-line no-console */
+		console.error( 'Error initiating Google login:', error );
```

Comment:
Do we wanna display the error in the console for all envs or just dev?

Reply:
Good point, we should remove that console.error